BrightID/src/components/GroupsScreens/GroupsScreen.js

```diff
@@ -128,4 +157,26 @@ const styles = StyleSheet.create({
   },
 });
 
-export default connect(({ groups }) => ({ ...groups }))(GroupsScreen);
+function mapStateToProps(state) {
+  let { groups } = state.groups;
+  const searchParam = state.groups.searchParam.toLowerCase();
+  const hasGroups = groups.length > 0;
+
+  // apply search filter to groups array
+  // NOTE: If below sorting/filtering gets too expensive at runtime use memoized selectors / reselect
+  if (searchParam !== '') {
+    groups = groups.filter((group) => {
+      if (group.type === PRIMARY_GROUP_TYPE) {
+        // since primary group does not have a name, match against the default name 'primary Group'
+        // this allows the user to explicitly search for the primary group
+        return PRIMARY_GROUP_NAME.includes(searchParam);
```

Comment:
Had the wrong assumption here that primary groups don't have a name. Just found out they do! Will update PR with a fix :)